include/simics/dmllib.h

```diff
@@ -631,6 +631,27 @@ _simple_event_predicate(lang_void *_data, lang_void *match_data) {
     return 0;
 }
 
+/*
+ * The modern serialized representation of after event data is a
+ * pseudo-dictionary, containing entries for method indices, arguments, and
+ * domains (associated objects for cancellation):
+ * [["indices", [index0, index1, ...]],
+ *  ["arguments", [arg0, arg1, ...]],
+ *  ["domains", [domain0, domain1, ...]]
+ * ]
+ * The entries may appear in any order, and any entry may be omitted if not
+ * applicable, (for example, a method call has no indices/arguments or
+ * associated objects). If all entries may be omitted, then Nil is used
+ * as the serialized representation.
+ *
+ * Before the introduction of domains, after event data was represented as
+ * follows:
+ * [[index0, index1, ...], [arg0, arg1, ...]]
+ * and when either list is empty, the other is represented directly; if both
+ * are empty, Nil is used.
+ * This is called the legacy representation of after event data, and is
+ * supported for backwards checkpoint compatibility.
```

Comment:
When documenting compatibility luggage, it's convenient for future archeologists if you mention approximate time ("early 2022"). Makes it easy to deduce if it's sufficiently ancient to be silently removed.

Reply:
Done.